### PR TITLE
Fix #41, add AlexNet to test suite

### LINKED ISSUE #41: AlexNet fails lint
```
(python3) [ezyang@devgpu014.ash3 ~/local/vision/torchvision/models] python alexnet.py                              
graph(%1, %2, %3, %4, %5, %6, %7, %8, %9, %10, %11, %12, %13, %14, %15, %16, %17) {
  %18.0 = CppOp[ConvForward](%17, %1, %2), uses = [[]];
  %20.0 = CppOp[ConvForward](%21, %3, %4), uses = [[]];
  %21 = Constant(), uses = [%20.i0];
  %23.0 = CppOp[ConvForward](%24, %5, %6), uses = [[]];
  %24 = Constant(), uses = [%23.i0];
  %26.0 = CppOp[ConvForward](%27, %7, %8), uses = [[]];
  %27 = Constant(), uses = [%26.i0];
  %29.0 = CppOp[ConvForward](%30, %9, %10), uses = [[]];
  %30 = Constant(), uses = [%29.i0];
  %32.0 = ^Transpose(0, 1)(%11), uses = [[%35.i2]];
  %34 = Constant(), uses = [%35.i1];
  %35.0 = ^Addmm(1, 1, False)(%12, %34, %32.0), uses = [[]];
  %37.0 = ^Transpose(0, 1)(%13), uses = [[%40.i2]];
  %39 = Constant(), uses = [%40.i1];
  %40.0 = ^Addmm(1, 1, False)(%14, %39, %37.0), uses = [[]];
  %42.0 = ^Transpose(0, 1)(%15), uses = [[%45.i2]];
  %44 = Constant(), uses = [%45.i1];
  %45.0 = ^Addmm(1, 1, False)(%16, %44, %42.0), uses = [[%0.i0]];
  return (%45.0);
}

(python3) [ezyang@devgpu014.ash3 ~/local/vision/torchvision/models] python alexnet.py 
Traceback (most recent call last):
  File "alexnet.py", line 67, in <module>
    model(x)
  File "/data/users/ezyang/pytorch/torch/nn/modules/module.py", line 224, in __call__
    result = self.forward(*input, **kwargs)
  File "/data/users/ezyang/pytorch/torch/jit.py", line 137, in forward
    tuple(self.parameters()) + flatten(args))
  File "/data/users/ezyang/pytorch/torch/jit.py", line 22, in record_trace
    torch._C._jit_pass_lint(trace)
RuntimeError: torch/csrc/jit/ir.cpp:266: lint: Assertion `in_scope.count(input) == 1` failed.
```

I'm investigating

### PR DESCRIPTION
**Pull request by ezyang**

### CODE REVIEW

File: torch/csrc/autograd/function.cpp [hunk 1]
```diff
@@ -41,9 +41,15 @@ void Function::createTrace(const variable_list& inputs, const variable_list& out
...
   auto& graph = state->graph;
-  auto* this_node = graph->appendNewNode<CppOp>(getSharedPtr());
+  // See Note [getValueTrace can allocate nodes]
+  std::vector<Node*> value_traces;
+  value_traces.reserve(inputs.size());
   for (auto& input: inputs) {
-    this_node->addInput(tracer::getValueTrace(state, input));
+    value_traces.emplace_back(tracer::getValueTrace(state, input));
+  }
+  auto* this_node = graph->appendNewNode<CppOp>(getSharedPtr());
```
Comment (zdevito) [inline torch/csrc/autograd/function.cpp:49]: Aside: this is another case where coupling node creation and node insertion is causing problems.

File: test/expect/TestJit.test_alexnet.expect [hunk 2]
```diff
@@ -0,0 +1,21 @@
+graph(%1, %2, %3, %4, %5, %6, %7, %8, %9, %10, %11, %12, %13, %14, %15, %16, %17) {
+  %18.0 = CppOp[ConvForward](%17, %1, %2), uses = [[]];
```
Comment (zdevito) [inline test/expect/TestJit.test_alexnet.expect]: What happened to the ReLU,MaxPool, and dropout operations?

**Review by zdevito — APPROVED** [review]
Looks good, assuming the AlexNet trace is correct for reasons I don't understand.

**Comment by apaszke** [comment]
Let's not merge it until we fix the ReLUs and other modules

**Pull request merged**